otter/util/fp.py

```diff
@@ -86,3 +86,16 @@ def assoc_obj(o, **k):
     new_o = deepcopy(o)
     new_o.__dict__.update(k)
     return new_o
+
+
+def take_while(pred, items):
+    """
+    Return list until pred is true for item in items. Similar to `takeWhile`
```

Comment:
Non-blocking - the `takeWhile` docstring seems pretty great :) Might be worth copying here - something like "Returns the longest prefix (possibly empty) of `items` of elements that satisfy `pred`"?